Add template weather forecast documentation (#42524)
home-assistant/home-assistant.github.io: source/_integrations/template.markdown

{% endconfiguration %}

### Weather Forecast data

> The weather forecast options should return a list of dictionaries, where each dictionary contains [forecast information](https://www.home-assistant.io/integrations/weather/#action-weatherget_forecasts) for the current timeframe. The data is slightly different for each forecast type: `hourly`, `daily`, and `twice_daily`.

Reviewer: ```suggestion
The weather forecast options return a list with the [forecast information](https://www.home-assistant.io/integrations/weather/#action-weatherget_forecasts) for the current timeframe. The data is slightly different for each forecast type: `hourly`, `daily`, and `twice_daily`.
```
Is it important that the user knows it returns a dictionary? Or could it just be called a list? Dictionary is software terminology.

Author: I think it's important because users need to build a list of dictionaries as the output.  They'll need to know the technical terms and even use Python constructs to create the output.

Author: I also think that if we don't use the term dictionary, people who know what they are doing won't know how to create the data output because it doesn't specify the data types.  And the people who don't know what they are doing still won't know what to do.  Adding the term dictionary only helps the situation.

Author: FYI we use the term dictionary all over the templating documentation as well

https://www.home-assistant.io/docs/configuration/templating/

Reviewer: I understand. Thank you for explaining!
